src/Data/Widget/Adapters/MetricAdapter.elm

```diff
@@ -45,16 +51,28 @@ valueForCell rows cell =
                 ""
 
 
-adapt : Dict String CellPosition -> Data -> ( String, String )
-adapt config data =
+adapt : AdapterConfig.Config -> Data -> ( String, String )
+adapt optionalConfig data =
     let
         rows =
             Array.fromList data.rows
 
+        sourceCell =
+            Dict.get "sourceCell" optionalConfig
+                |> Maybe.withDefault (CellPosition.asJsonValue sourceCellPosition)
+                |> Json.decodeValue CellPosition.decoder
+                |> Result.withDefault sourceCellPosition
+
```

Comment:
So if we have a `Value` defined then decode it, otherwise we take the default `CellPosition`, encode it to be a `Value` which can then be passed to the decoder. I think I understand!